lib/fathead/celery/parse.js

```diff
@@ -0,0 +1,121 @@
+	var newArt = article.slice(article.lastIndexOf(".") + 1, article.length);
+	/_/g.test(article) ? redirs.push(newArt.replace(/_/g, " ")) : redirs.push(newArt);
+	return redirs.map(function (art) {
+		return art + "\tR\t" + article + "\t\t\t\t\t\t\t\t\t\t\n";
+	});
+}
+
+function getDocs(celeryData) {
+	var methods = [];
+	var $ = cheerio.load(celeryData);
+	var $docs = $("#api-reference").children(".toctree-wrapper").children(".section");
+	for (var i = 3; i < $docs.length; i += 1) {
+		var $elem = $docs.eq(i);
+		if (!$elem.children(".class").html()) continue;
+		var id = $elem.attr("id").replace(/-/g, ".");
+		methods.push({
+			category: id + " Functions",
+			baseURL: id + ".html",
+			$methods: $elem.children(".class").children("dd").children()
+		});
+	}
+	return methods;
+}
+
+function getDetails(_ref2) {
+	var category = _ref2.category,
+	    baseURL = _ref2.baseURL,
+	    $methods = _ref2.$methods;
+
+	var details = [];
+	baseURL = "http://docs.celeryproject.org/en/stable/reference/" + baseURL;
+	for (var i = 0; i < $methods.length; i += 1) {
+		var $elem = $methods.eq(i);
+		if (!/method/g.test($elem.attr("class"))) continue;
+		var article = $elem.find(".descclassname").text() + $elem.find(".descname").text();
+		var url = baseURL + $elem.find(".headerlink").attr("href");
+		var abstract = makeAbstract($elem);
+		details.push({ article: article, category: category, url: url, abstract: abstract });
+	}
+	return details;
+}
+
+function formatDescription($elem) {
+	if (!$elem.html()) return "";
+	var desc = [];
+	$elem.map(function (i) {
+		var attr = $elem.eq(i).attr("class");
+		if (attr === "rubric") return desc.push("<span class=\"prog__sub\">" + $elem.eq(i).text() + "</span>");
+		desc.push("<p>" + $elem.eq(i).text() + "</p>");
+	});
+	return desc.join("");
+}
+
+function summaryHelper($tr) {
+	var $ul = $tr.find("ul");
+	if (!$ul.html()) return "<p>" + $tr.children("td").text() + "</p>";
+	return '<ul class="prog__ul">' + $ul.text().split("\n").filter(function (curr) {
+		return curr ? curr : null;
+	}).map(function (curr) {
+		return "<li>" + curr + "</li>";
+	}).join("") + "</ul>";
+}
+
+function formatSummary($elem) {
+	var summary = [];
+	for (var i = 0; i < $elem.length; i += 1) {
+		var $tr = $elem.eq(i);
+		var name = "<span class=\"prog__sub\">" + $tr.children("th").text().replace(/:/g, "") + "</span>";
+		var body = summaryHelper($tr);
+		summary.push(name + "\n" + body);
+	}
+	return summary.join("");
+}
+
+function formatDom(desc) {
+	return desc.replace(/<\s*a[^>]*>/ig, "").replace(/<\s*\/\s*a>/ig, "").replace(/\[source\]¶/g, "").replace(/¶/, "").replace(/\n/g, "");
+}
+
+function makeAbstract($elem) {
+	var abstractData = {
+		startDom: "<section class=\"prog__container\">",
+		title: "<pre><code>" + $elem.find("dt").text() + "</code></pre>",
+		description: formatDescription($elem.children("dd").children("p")),
+		example: ("<pre><code>" + $elem.find(".highlight pre").text() + "</code></pre>").split("...").join("\n"),
```

Comment:
We need to ensure we have text to put in the code block.

The output contains 644 empty code blocks. E.g. https://beta.duckduckgo.com/253/AsyncResult?ia=about&iax=1

![python__new_celery_fathead_by_lord-lavios_ _pull_request__827_ _duckduckgo_zeroclickinfo-fathead](https://cloud.githubusercontent.com/assets/873785/23312283/2e0f0e0e-fa87-11e6-85f0-eb6c03672471.png)